CMRF/Core/AbstractCall.php

```diff
@@ -88,7 +88,9 @@ public function getValues() {
   }
 
   public function getHash() {
-    $request = $this->getRequest();
+    // Include Entity and Action in the request hash to ensure cached dataset retrieval 
+    // actually returns the expected values
+    $request = [ $this->getEntity(), $this->getAction(), $this->getRequest() ];
```

Comment:
As this invalidates all currently cached calls, should we add a compatibility layer?

Reply:
Here's a case in point:

![image](https://github.com/CiviMRF/CMRF_Abstract_Core/assets/32186614/f938f353-d7d4-436a-8b24-087a3927019c)

These two API calls return the same data, because $params is exactly the same. Every existing cached call should probably be updated anyway, or they're going to run into this issue. All the more if they're set for a long period!

Reply:
Makes sense, so invalidating the Cache is intended here.